From e1252cbf65001487ff3b94bbc141acacf6e528b6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 16 May 2012 20:03:34 +0200
Subject: [PATCH] configure: add option to disable libelf support

Upstream-Status: Backport 
Applied in master
http://git.gnome.org/browse/glib/commit/?id=9e1d4da14d8e69fb60da1a9733ac1d58d0dcd8b2
can be dropped when upgrading to 2.34.* or newer

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 configure.ac |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index ed4c10e..210df90 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1815,6 +1815,9 @@ AC_SUBST(XATTR_LIBS)
 dnl ************************
 dnl *** check for libelf ***
 dnl ************************
+AC_ARG_ENABLE(libelf,
+              AC_HELP_STRING([--disable-libelf], [build without libelf support]))
+if test "x$enable_libelf" != "xno"; then
 
 PKG_CHECK_MODULES([LIBELF], [libelf >= 0.8.12], [have_libelf=yes], [have_libelf=maybe])
 
@@ -1835,6 +1838,7 @@ fi
 if test $have_libelf = yes; then
   AC_DEFINE(HAVE_LIBELF, 1, [Define if libelf is available])
 fi
+fi
 
 dnl ****************************************
 dnl *** platform dependent source checks ***
-- 
1.7.8.6

